app/assets/javascripts/dialog-polyfill.js

```diff
@@ -0,0 +1,533 @@
+(function() {
```

Comment:
@sic-f did you write all this code?  What's the source of it?  should it be in the vendor dir with a license?